tapdb/universe.go

```diff
@@ -233,27 +233,36 @@ func upsertAssetGen(ctx context.Context, db UpsertAssetStore,
 
 	// Finally, if there's a group key associated with the asset, then
 	// we'll insert that now as well.
-	if groupKey != nil && genesisProof.GroupKeyReveal == nil {
-		fmt.Printf("nil group reveal!")
-	}
-
-	if groupKey != nil && genesisProof.GroupKeyReveal != nil {
-		reveal := genesisProof.GroupKeyReveal
-		rawKey, err := reveal.RawKey.ToPubKey()
-		if err != nil {
-			return 0, err
-		}
-
+	if groupKey != nil {
+		// Every group-related issuance must be accompanied by a group
+		// witness.
+		groupWitness := genesisProof.Asset.PrevWitnesses[0].TxWitness
```

Comment:
Should we Squash the changes in this file with 42f097ac to make the review flow somewhat easier? Since this isn't really related to storing the group anchors early, just storing them correctly?